### ISSUE TITLE
EvaporativeCooler objects lack required fields and defaults

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#5060]
All of the EvaporativeCooler:\* objects lack required-field and default tags (except one type has a required Name field).  Presumably all of the Name and Node Name fields should be required, along with possible others.  All flow rates and capacities should default to autosize.  

This came to light from troubleshooting a run with EVAPORATIVECOOLER:INDIRECT:RESEARCHSPECIAL objects where the Secondary Fan Flow Rate was autosizing to zero.  The Primary Design Air Flow Rate had been left blank by mistake which produced no warnings, just a zero flow rate which then carried into the autosizing of the secondary flow rate.

**Comment by mjwitte** [comment]
Also, EvaporativeCooler:Direct:ResearchSpecial does not have a field for Primary Air Flow Rate or Pad Area, so how can the pump autosize?
  N3 , \field Water Pump Power Sizing Factor
       \type real
       \units W/(m3/s)
       \note this field is used when the previous field is set to autosize. The pump power is scaled with Primary Air Design Air Flow Rate

**Comment by mjwitte** [comment]
So, adding Autosize to the Primary Design Air Flow Rate fields in EvaporativeCooler:Indirect:ResearchSpecial is still resulting in zero for autosized secondary air flow rate, and the primary flow rate is not reporting a size.  Appears that sizing is broken in this particular evap cooler.  Example file DirectIndirectEvapCoolers also has autosized EvaporativeCooler:Indirect:ResearchSpecial objects.  These do not report a primary flow rate size, but they do report non-zero secondary flow rates.  (Which may help give a clue to other problems in the original problem file.)  The example file has been added as the defect file for this.
_Making a new issue for the sizing problem._ #5061

**Comment by Nigusse** [comment]
@mjwitte @Myoldmopar In "EvaporativeCooler:Direct:ResearchSpecial" there is no primary air design flow rate input field.  Mike, as you pointed out the "Water Pump Power Sizing Factor" is not helpful without the primary air flow rate.  Existing code will size the recirculation pump power to zero if the "Recirculating Water Pump Design Power" input field is  set autosized. All example files will require transition with auto-size as value if we add new input field.

Shall I go add input field for Primary Air Design Air Flow Rate for this object?

**Comment by Myoldmopar** [comment]
I'm still OK with transition changes during this period, so don't let that stop this fix if it is needed.

**Comment by mjwitte** [comment]
@EnergyArchmage Can you comment on this?  @Nigusse How does EvaporativeCooler:Direct:ResearchSpecial function without an air flow rate?  Is the Primary Air Design Flow Rate in EvaporativeCooler:InDirect:ResearchSpecial used for anything other than sizing other parts of that unit?

**Comment by Myoldmopar** [comment]
:zap: :dash: :zap: :droplet: :cloud:

**Comment by Nigusse** [comment]
Runs using the inlet flow rate during initialization. See code snippet below:

```
    // set the volume flow rates from the input mass flow rates
    EvapCond( EvapCoolNum ).VolFlowRate = Node( InletNode ).MassFlowRate / RhoAir;
```

**Comment by Nigusse** [comment]
Actually the design volume flow rate is saved in the variable "EvapCond( EvapCoolNum ).VolFlowRate" and is overridden every time step.  The design flow rate is not saved and will not be available once reported.

**Comment by Nigusse** [comment]
@mjwitte @Myoldmopar I checked the IDD, all lacking required-field and default tags have been fixed/added. Also a primary air flow rate field has been added to EvaporativeCooler:Direct:ResearchSpecial object and the autosizing calculation runs ok. 

What is missing for "Primary Air Design Flow Rate" input field of this object is that if the field is is left blank, it will never throw any error message and the simulation runs to completions. Do we need to catch and provide warning if this field is left blank?  

I think we have noted that the evap coolers use the parent supply air flow rate, so the evap cooler runs successfully even if the Primary Air Design Flow Rate is zero or not sized.

**Comment by mjwitte** [comment]
@Nigusse For both EvaporativeCooler:Direct:ResearchSpecial and EvaporativeCooler:Inirect:ResearchSpecial the Primary/Secondary Air Design Flow Rate fields are autosizeable, not required, and have no default.  That says to me if they are left blank, then they are zero.  If they don't enforce that and take the flow that's sent to them, then I suppose it's OK, but it is a bit odd for that to zero - will other autosized fields like the fan power and pump power end up being zero if these are left blank?  Perhaps the air flow rate fields should default to autosize?

**Comment by Nigusse** [comment]
@mjwitte Yes, if the Primary Air Design Flow Rate fields left blank, then both flow rate and pump power are sized to zero.  This is for EvaporativeCooler:Direct:ResearchSpecial.  The indirect Evap cooler pump power is scaled using the secondary air flow rate.  But the logic is the same.

**Comment by mjwitte** [comment]
Then it seems we should default those to \default autosize unless that breaks something.

**Comment by Nigusse** [comment]
I will default the secondary fan and pump power input fields to autosize. I also see the  for direct special evap cooler the field "Water Pump Power Sizing Factor" is set to zero if left blank in get input even if it has a default value. This also has to fixed.  I will make these changes.

**Comment by mjwitte** [comment]
Closed via #5136 and #5393
